x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/TextEmbedding/MultilingualE5SmallServiceSettings.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ *
+ * this file was contributed to by a generative AI
+ */
+
+package org.elasticsearch.xpack.inference.services.TextEmbedding;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+import org.elasticsearch.xpack.inference.services.settings.MlNodeDeployedServiceSettings;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.TransportVersions.ML_TEXT_EMBEDDING_INFERENCE_SERVICE_ADDED;
+import static org.elasticsearch.xpack.inference.services.TextEmbedding.TextEmbeddingService.MULTILINGUAL_E5_SMALL_MODEL_ID;
+import static org.elasticsearch.xpack.inference.services.TextEmbedding.TextEmbeddingService.MULTILINGUAL_E5_SMALL_MODEL_ID_LINUX_X86;
+
+public class MultilingualE5SmallServiceSettings extends TextEmbeddingServiceSettings {
+
+    public static final String NAME = "multilingual_e5_small_service_settings";
+    public static final List<String> MODEL_VARIANTS = List.of(MULTILINGUAL_E5_SMALL_MODEL_ID, MULTILINGUAL_E5_SMALL_MODEL_ID_LINUX_X86);
+
+    public MultilingualE5SmallServiceSettings(int numAllocations, int numThreads, String modelVariant) {
+        super(numAllocations, numThreads, modelVariant);
+    }
+
+    /**
+     * Parse the MultilingualE5SmallServiceSettings from map and validate the setting values.
+     *
+     * If required setting are missing or the values are invalid an
+     * {@link ValidationException} is thrown.
+     *
+     * @param map Source map containing the config
+     * @return The {@code MultilingualE5SmallServiceSettings} builder
+     */
+    public static MultilingualE5SmallServiceSettings.Builder fromMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+        Integer numAllocations = ServiceUtils.removeAsType(map, NUM_ALLOCATIONS, Integer.class);
+        Integer numThreads = ServiceUtils.removeAsType(map, NUM_THREADS, Integer.class);
+
+        validateParameters(numAllocations, validationException, numThreads);
+
+        String version = ServiceUtils.removeAsType(map, MODEL_VERSION, String.class);
+        if (version != null) {
+            if (MODEL_VARIANTS.contains(version) == false) {
+                validationException.addValidationError("unknown Multilingual-E5-Small model version [" + version + "]");
```

Comment:
I'm happy to add that if you think it would be valuable. I think this exception will be unlikely to be seen in production once I introduce my `CustomElandModel` change.